### PR TITLE
Allow ember-data to work when `host` is not defined

### PR DESCRIPTION
**Pull request by danmcclain**
This uses the headers/request fron express to figure out where to make
the najax request to.

This is in lieu of a proper jQuery AJAX transport

### CODE REVIEW

File: app/initializers/fastboot/ajax.js [hunk 1]
```diff
@@ -1,6 +1,17 @@
 /* globals najax */
+import Ember from 'ember';
+
+const { get } = Ember;
 
 var nodeAjax = function(options) {
+  let httpRegex = /^https?:\/\//
+
+  if (!httpRegex.test(options.url)) {
+    let host = `${get(this, 'fastboot._fastbootInfo.reques.protocol')}://${get(this, 'fastboot.headers.host')}`;
```
Comment (tomdale) [inline app/initializers/fastboot/ajax.js]: I have a doubt

Comment (tomdale) [inline app/initializers/fastboot/ajax.js]: I think we need tests for this. :trollface:

**Comment by tomdale** [comment]
This is also needed by #126; we should think about exposing this info as public API to support @paulelliott's use case as well.

**Comment by danmcclain** [comment]
@tomdale Added `host` to the service, PR to `ember-fastboot-server`: https://github.com/ember-fastboot/ember-fastboot-server/pull/25

**Comment by paulelliott** [comment]
This would solve my issues, for sure. Any ETA on when we could see this merged?

**Comment by danmcclain** [comment]
@paulelliott #136 replaces this PR in terms of introducing the host API. Once #136 is merged, I'm going to update this issue to use the new host API

**Comment by danmcclain** [comment]
@tomdale this is rebased, and should be good to merge

**Comment by tomdale** [comment]
@danmcclain This should probably have a better error message if there isn't a host whitelisted, right? People randomly using Ember Data will hit it and the error message about accessing `fastboot.host` will probably be very confusing.

**Comment by danmcclain** [comment]
@tomdale Added a long winded, informative error message :smile:

File: app/initializers/fastboot/ajax.js [hunk 2]
```diff
@@ -1,6 +1,19 @@
...
+const { get } = Ember;
 
 var nodeAjax = function(options) {
+  let httpRegex = /^https?:\/\//
+
+  if (!httpRegex.test(options.url)) {
+    try {
+      options.url = get(this, 'fastboot.host') + options.url
+    } catch (fbError) {
+      throw new Error('You are using Ember Data with no host defined in your adapter. This will attempt to use the host of the FastBoot request, which is not configured for the current host of this request. Please set the hostWhitelist property for in your environement.js. FastBoot Error: ' + fbError.message);
```
Comment (acorncom) [inline app/initializers/fastboot/ajax.js]: Right at the very end of this long error message `environment` is mis-spelled :-)

Comment (danmcclain) [inline app/initializers/fastboot/ajax.js]: Thanks!

**Comment by danmcclain** [comment]
@tomdale Ping

**Comment by tomdale** [comment]
Mérgèd

**Pull request merged**